src/components/Modals/CancelMatchModal.tsx

```diff
@@ -0,0 +1,246 @@
+                  </DialogModalBase.Description>
+                </div>
+              )}
+
+              {supportSuccessful === false && (
+                <DialogModalBase.Form>
+                  {ownType === 'student' ? (
+                    <DialogModalBase.Subheading>
+                      Warum konntest du {matchFirstname} nicht unterstützen?
+                    </DialogModalBase.Subheading>
+                  ) : (
+                    <DialogModalBase.Subheading>
+                      Warum konnte dich {matchFirstname} nicht unterstützen?
+                    </DialogModalBase.Subheading>
+                  )}
+                  <DialogModalBase.InputCompound direction="vertical">
+                    {Object.entries(reasonOptions).map(([key, value]) => (
+                      <DialogModalBase.CheckboxSingle
+                        key={key}
+                        label={value}
+                        value={key}
+                        selected={noHelpReason}
+                        onSelect={setNoHelpReason}
+                      />
+                    ))}
+                  </DialogModalBase.InputCompound>
+                </DialogModalBase.Form>
+              )}
+              {!dissolved &&
+                supportSuccessful != null &&
+                (supportSuccessful ||
+                  (!supportSuccessful && noHelpReason != null)) && (
+                  <DialogModalBase.ButtonBox>
+                    <DialogModalBase.Button
+                      onClick={() =>
+                        endCollaboration().then(() => setDissolved(true))
+                      }
+                      label="Auflösen"
+                    />
+                  </DialogModalBase.ButtonBox>
+                )}
+            </DialogModalBase.Content>
+          </div>
+        ) : (
+          <DialogModalBase.Content>
+            <DialogModalBase.Description>
+              Die Zusammenarbeit wurde erfolgreich beendet!
+            </DialogModalBase.Description>
+
+            <DialogModalBase.Form>
+              <DialogModalBase.Subheading>
+                {projectCoaching &&
+                  `Möchtest du mit einem neuen ${
+                    ownType === 'student' ? 'Coachee' : 'Coach'
+                  } verbunden werden?`}
+                {!projectCoaching &&
+                  'Möchtest du mit einem/einer neuen Lernpartner*in verbunden werden?'}
+              </DialogModalBase.Subheading>
+
+              <DialogModalBase.InputCompound direction="horizontal">
+                <DialogModalBase.CheckboxSingle
+                  value
+                  selected={newMatchWanted}
+                  onSelect={setNewMatchWanted}
+                  label="Ja"
+                />
+                <DialogModalBase.CheckboxSingle
+                  value={false}
+                  selected={newMatchWanted}
+                  onSelect={setNewMatchWanted}
+                  label="Nein"
+                />
```

Comment:
Maybe a confirm button here would be a good idea. Or alternatively buttons for "Ja"or "Nein". From my user perspective I'm not convinced before leaving the modal that my choice will persist.
![Bildschirmfoto 2020-12-05 um 03 59 54](https://user-images.githubusercontent.com/33378936/101231942-d6bf4200-36ae-11eb-897c-60cb9ed4d2dd.png)

Reply:
Fixed in a1002b88b4f361ff88fa73cfd0284bbee2015483
![Screen Shot 2020-12-05 at 11 06 41](https://user-images.githubusercontent.com/38134006/101239736-3edf4980-36ea-11eb-9d5f-d20d5f0f517f.png)